Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1367,6 +1456,18 @@ private static void OnItemsSourcePropertyChanged(DependencyObject d, DependencyP
                 // can be set when the DataGrid is not part of the visual tree.
                 dataGrid._measured = false;
                 dataGrid.InvalidateMeasure();
+
+                // Determine if incremental loading should be used
+                if (dataGrid.ItemsSource is ISupportIncrementalLoading incrementalItemsSource)
+                {
+                    dataGrid._incrementalItemsSource = incrementalItemsSource;
+                    dataGrid._loadingItems = false;
```

Comment:
Should cancel the existing request for more items rather than setting this to false.